### PR TITLE
Try to fix Sentry crash on bad B-spline curve [buildNum:2024.02.21.18]

### PR DESCRIPTION
**Pull request by dassaf4**
Hopefully resolves https://github.com/iTwin/imodel-native-internal/issues/388 with paranoid validation.

Without a test case to reproduce this Sentry crash, we don't know how this corrupt B-spline curve is creeping into tile generation.

### CODE REVIEW

**Review by saeeedtorabi — APPROVED** [review]

**Review by MichaelBelousov — APPROVED** [review]

File: iModelCore/GeomLibs/geom/test/CurvePrimitiveTest/t_IGeometry.cpp [hunk 1]
```diff
@@ -1010,6 +1010,13 @@ TEST(FlatBuffer, IsNanValues)
     curvePtr3->weights[2] = 1.0;
     Check::True(validator->IsValidGeometry(curvePtr3));
 
+    MSBsplineCurve myCurve;
+    auto cv = ICurvePrimitive::CreateBsplineCurve(curvePtr1);
+    Check::True(cv->GetMSBsplineCurve(myCurve), "GetMSBsplineCurve validates valid curve");
+    myCurve.knots = nullptr;    // invalidate it
+    cv = ICurvePrimitive::CreateBsplineCurveSwapFromSource(myCurve);
+    Check::False(cv->GetMSBsplineCurve(myCurve), "GetMSBsplineCurve validates invalid curve");
```
Comment (saeeedtorabi) [inline iModelCore/GeomLibs/geom/test/CurvePrimitiveTest/t_IGeometry.cpp:1018]: How about "GetMSBsplineCurve invalidates invalid curve"?

Comment (dassaf4) [reply iModelCore/GeomLibs/geom/test/CurvePrimitiveTest/t_IGeometry.cpp:1018]: D'oh! I'll fix.

**Comment by dassaf4** [comment]
Update: Sentry reports another [crash](https://bentley-systems-inc.sentry.io/issues/5144807837) with the same callstack. :(

**Comment by saeeedtorabi** [comment]
> Update: Sentry reports another [crash](https://bentley-systems-inc.sentry.io/issues/5144807837) with the same callstack. :(

What?! Does Sentry get the latest code? or is it using an older release?

**Comment by dassaf4** [comment]
> Does Sentry get the latest code? or is it using an older release?

Presumably it crashed with this change. I see a later version of iModelJsNativeVersion in the Sentry report at least.

**Pull request merged**